### DISCUSSION TITLE
MWP on Windows using cygwin

### DISCUSSION THREAD

**Discussion by underthebanyan** [stronnag/mwptools#85] [Q&A] [answered]
I have a very foggy memory of programming and linux from uni, and yet I've managed to follow the vast majority of instructions all until the runtime section. I was able to edit .bashrc with emacs and insert and save the provided commands. I have mobaxterm server open, but when I type in mwp in cygwin, it says: Unable to init server could not connect to 127.0.0.1; Connection refused

I don't quite understand how mobaxterm and cygwin are supposed to connect, but in the terminal in mobaxterm, the second line does say "Your DISPLAY is set to 192.168 yadda yadda."

I'm stuck here. If anyone knows what I'm missing I'd love to learn, thanks.

**Comment by stronnag** [comment, marked as answer]
I'm not sure if mobaxterm's behaviour (or some underlying WIndows behaviour) has changed; it no  longer starts an X-Server on :1 (loopback address), but rather it seems X is only on the address of the NIC. So you must tell mwp about this.

* In a Win10 VM, when I start MobaXterm it tells me "Your DISPLAY is set to 172.31.0.194:1.0" 
* In a cygwin terminal:
```
    $  DISPLAY=172.31.0.194:1.0 mwp
```
And mwp opens as expected. If you always have the same assigned IP, you could add the DISPLAY variable to `.bashrc` so it's always set
```
export DISPLAY=172.31.0.194:1.0
``` 
(for your fixed address of course).

**Reply by underthebanyan** [reply]
Excellent, thank you so much. Worked like a charm. I aspire to code like you one day
